src/prisma/cli/_node.py

```diff
@@ -0,0 +1,320 @@
+            raise RuntimeError(
+                'Could not install nodeenv to the expected directory; See the output above for more details.'
+            )
+
+        # TODO: what hapens on cygwin?
+        if platform.name() == 'windows':
+            bin_dir = cache_dir / 'Scripts'
+            if target == 'node':
+                path = bin_dir / 'node.exe'
+            else:
+                path = bin_dir / f'{target}.cmd'
+        else:
+            path = cache_dir / 'bin' / target
+
+        if target == 'npm':
+            return cls(path=path, resolver='nodeenv')
+        elif target == 'node':
+            return cls(path=path, resolver='nodeenv')
+        else:
+            raise UnknownTargetError(target=target)
+
+
+class NodeJSPythonStrategy(Strategy):
+    target: Target
+    resolver: Literal['nodejs-bin']
+
+    def __init__(self, *, target: Target) -> None:
+        self.target = target
+        self.resolver = 'nodejs-bin'
+
+    def __run__(
+        self,
+        *args: str,
+        check: bool = False,
+        cwd: Path | None = None,
+        stdout: File | None = None,
+        stderr: File | None = None,
+        env: Mapping[str, str] | None = None,
+    ) -> subprocess.CompletedProcess[bytes]:
+        func = None
+        if self.target == 'node':
+            func = nodejs.node.run
+        elif self.target == 'npm':
+            func = nodejs.npm.run
+        else:
+            raise UnknownTargetError(target=self.target)
+
+        # TODO
+        return func(  #  type: ignore
+            args,
+            check=check,
+            cwd=cwd,
+            env=env,
+            stdout=stdout,
+            stderr=stderr,
+        )
+
+    @property
+    def target_bin(self) -> Path:
+        return Path(nodejs.node.path).parent
+
+
+Node = Union[NodeJSPythonStrategy, NodeBinaryStrategy]
+
+
+def resolve(target: Target) -> Node:
+    if target not in {'node', 'npm'}:
+        raise UnknownTargetError(target=target)
+
+    if config.use_nodejs_bin:
+        log.debug('Checking if nodejs-bin is installed')
+        if nodejs is not None:
+            log.debug('Using nodejs-bin with version: %s', nodejs.node_version)
+            return NodeJSPythonStrategy(target=target)
+
+    return NodeBinaryStrategy.resolve(target)
+
+
+def _update_path_env(
+    *,
+    env: Mapping[str, str] | None,
+    target_bin: Path,
+) -> dict[str, str]:
+    """Returns a modified version of `os.environ` with the `PATH` environment variable updated
+    to include the location of the downloaded Node binaries.
+    """
+    if env is None:
+        env = dict(os.environ)
+
+    assert target_bin.exists(), 'Target `bin` directory does not exist'
+
+    path = env.get('PATH', '') or os.environ.get('PATH', '')
+    if path:
+        # handle the case where the PATH already ends with the `:` separator (this probably shouldn't happen)
+        if path.endswith(':'):
+            path = f'{path}{target_bin.absolute()}'
```

Comment:
Just wondering if we should prefix our node version vs adding it to the end... That would give more control over our code to a known `node` or `npm` version that we have brought in ourselves via the `prisma-py[node]` install method. Probably a lot less room for weird bugs that way. At least on my system, this would mean even though I installed the `prisma[node]` way, it would still resolve to whatever my global `node` was which would sit earlier on the PATH

Reply:
Good point, I'll update this, thanks!